### PR TITLE
feat: Flexible Button variants

### PR DESCRIPTION
**Pull request by Sodik**
Replace property kind by combination of variant and color properties. Property kind will be removed later.

### CODE REVIEW

File: packages/ui/__stories__/Button.stories.tsx [hunk 1]
```diff
@@ -156,3 +156,53 @@ export const LinkSemantics = () => {
     </>
   )
 }
+
+export const AllVariants = () => {
```
Comment (danad02) [inline packages/ui/__stories__/Button.stories.tsx:160]: Can we just update the `Template` with all combinations of `variant` and `color`? Other stories are bound to the template and will update accordingly.

File: packages/ui/__stories__/Button.stories.tsx [hunk 2]
```diff
@@ -156,3 +156,53 @@ export const LinkSemantics = () => {
...
+
+        <Button {...props} variant="outlined" color="primary" />
+        <Button {...props} iconLeft={Info} iconLeftAriaLabel={'Additional information'} variant="outlined" color="primary" />
+        <Button {...props} variant="outlined" color="secondary" />
+        <Button {...props} iconLeft={Info} iconLeftAriaLabel={'Additional information'} variant="outlined" color="secondary" />
+
+        <Button {...props} variant="text" color="primary" />
+        <Button {...props} iconLeft={Info} iconLeftAriaLabel={'Additional information'} variant="text" color="primary" />
+        <Button {...props} variant="text" color="secondary" />
+        <Button {...props} iconLeft={Info} iconLeftAriaLabel={'Additional information'} variant="text" color="secondary" />
```
Comment (danad02) [inline packages/ui/__stories__/Button.stories.tsx:181]: Is there a difference between `variant="text" color="primary"` and `variant="text" color="secondary"`?

**Review by danad02 — CHANGES_REQUESTED** [review]
I like this change of props! Leaving two outstanding comments.

File: packages/ui/__stories__/Button.stories.tsx (hunk 2 above)
Comment (Sodik) [reply packages/ui/__stories__/Button.stories.tsx:181]: Text inherits color so there is no deference. Maybe we will need to change it

**Review by danad02 — APPROVED** [review]
LGTM :+1:

**Pull request merged**